File: docs/CartApiClient.java
// CartApiClient.java - Android Java implementation for POS Cart API
package com.candykush.pos.api;

import okhttp3.*;
import com.google.gson.Gson;
import java.io.IOException;
import java.util.concurrent.TimeUnit;

public class CartApiClient {
    private static final String BASE_URL = "https://pos-candy-kush.vercel.app/api";
    private final OkHttpClient client;
    private final Gson gson;

    public CartApiClient() {
        this.client = new OkHttpClient.Builder()
            .connectTimeout(10, TimeUnit.SECONDS)
            .readTimeout(10, TimeUnit.SECONDS)
            .writeTimeout(10, TimeUnit.SECONDS)
            .build();
        this.gson = new Gson();
    }

    // Get current cart contents
    public Cart getCart() throws IOException {
        Request request = new Request.Builder()
            .url(BASE_URL + "/cart")
            .get()
            .addHeader("Content-Type", "application/json")
            .build();

        try (Response response = client.newCall(request).execute()) {
            if (response.isSuccessful()) {
                String json = response.body().string();
                CartResponse cartResponse = gson.fromJson(json, CartResponse.class);
                if (cartResponse.isSuccess()) {
                    return cartResponse.getCart();
                }
            }
        }
        return null;
    }

    // Get payment status
    public PaymentStatus getPaymentStatus() throws IOException {
        Request request = new Request.Builder()
            .url(BASE_URL + "/cart/payment")
            .get()
            .addHeader("Content-Type", "application/json")
            .build();

        try (Response response = client.newCall(request).execute()) {
            if (response.isSuccessful()) {
                String json = response.body().string();
                PaymentResponse paymentResponse = gson.fromJson(json, PaymentResponse.class);
                if (paymentResponse.isSuccess()) {
                    return paymentResponse.getPaymentStatus();
                }
            }
        }
        return null;
    }

    // Data classes (same structure as Kotlin version)
    public static class CartItem {
        private String id;
        private String productId;
        private String name;
        private int quantity;
        private double price;
        private double total;
        private Double weight;
        private String unit;

        // Getters and setters
        public String getId() { return id; }
        public void setId(String id) { this.id = id; }
        // ... add other getters/setters
    }

    public static class Customer {
        private String id;
        private String name;
        private String phone;

        // Getters and setters
        public String getName() { return name; }
        public void setName(String name) { this.name = name; }
        // ... add other getters/setters
    }

    public static class Cart {
        private java.util.List<CartItem> items;
        private Discount discount;
        private Tax tax;
        private Customer customer;
        private String notes;
        private double total;
        private String lastUpdated;

        // Getters and setters
        public java.util.List<CartItem> getItems() { return items; }
        public void setItems(java.util.List<CartItem> items) { this.items = items; }
        public double getTotal() { return total; }
        public void setTotal(double total) { this.total = total; }
        public Customer getCustomer() { return customer; }
        public void setCustomer(Customer customer) { this.customer = customer; }
        // ... add other getters/setters
    }

    public static class PaymentStatus {
        private String status;
        private String timestamp;
        private double amount;
        private String method;
        private String transactionId;

        // Getters and setters
        public String getStatus() { return status; }
        public void setStatus(String status) { this.status = status; }
        public double getAmount() { return amount; }
        public void setAmount(double amount) { this.amount = amount; }
        // ... add other getters/setters
    }

    // Response wrapper classes
    public static class CartResponse {
        private boolean success;
        private Cart cart;
        private String timestamp;

        public boolean isSuccess() { return success; }
        public Cart getCart() { return cart; }
    }

    public static class PaymentResponse {
        private boolean success;
        private PaymentStatus paymentStatus;

        public boolean isSuccess() { return success; }
        public PaymentStatus getPaymentStatus() { return paymentStatus; }
    }

    // Supporting classes
    public static class Discount {
        private String type;
        private double value;
        // getters/setters
    }

    public static class Tax {
        private double rate;
        private double amount;
        // getters/setters
    }
}

File: docs/CartApiClient.kt
// CartApiClient.kt - Android Kotlin implementation for POS Cart API
package com.candykush.pos.api

import okhttp3.OkHttpClient
import okhttp3.Request
import com.google.gson.Gson
import kotlinx.coroutines.*
import java.io.IOException
import java.util.concurrent.TimeUnit

data class CartItem(
    val id: String,
    val productId: String,
    val name: String,
    val quantity: Int,
    val price: Double,
    val total: Double,
    val weight: Double? = null,
    val unit: String? = null
)

data class Customer(
    val id: String? = null,
    val name: String? = null,
    val phone: String? = null
)

data class Discount(
    val type: String, // "percentage" or "fixed"
    val value: Double
)

data class Tax(
    val rate: Double,
    val amount: Double
)

data class Cart(
    val items: List<CartItem>,
    val discount: Discount,
    val tax: Tax,
    val customer: Customer?,
    val notes: String,
    val total: Double,
    val lastUpdated: String?
)

data class CartResponse(
    val success: Boolean,
    val cart: Cart,
    val timestamp: String
)

data class PaymentStatus(
    val status: String, // "idle", "processing", "completed", "failed"
    val timestamp: String?,
    val amount: Double,
    val method: String?,
    val transactionId: String?
)

data class PaymentResponse(
    val success: Boolean,
    val paymentStatus: PaymentStatus
)

class CartApiClient(
    private val baseUrl: String = "https://pos-candy-kush.vercel.app/api",
    private val gson: Gson = Gson()
) {
    private val client = OkHttpClient.Builder()
        .connectTimeout(10, TimeUnit.SECONDS)
        .readTimeout(10, TimeUnit.SECONDS)
        .writeTimeout(10, TimeUnit.SECONDS)
        .build()

    // Get current cart contents
    suspend fun getCart(): Cart? = withContext(Dispatchers.IO) {
        try {
            val request = Request.Builder()
                .url("$baseUrl/cart")
                .get()
                .addHeader("Content-Type", "application/json")
                .build()

            val response = client.newCall(request).execute()
            val json = response.body?.string()

            if (response.isSuccessful && json != null) {
                val cartResponse = gson.fromJson(json, CartResponse::class.java)
                if (cartResponse.success) {
                    return@withContext cartResponse.cart
                }
            }
            null
        } catch (e: IOException) {
            e.printStackTrace()
            null
        }
    }

    // Get payment status
    suspend fun getPaymentStatus(): PaymentStatus? = withContext(Dispatchers.IO) {
        try {
            val request = Request.Builder()
                .url("$baseUrl/cart/payment")
                .get()
                .addHeader("Content-Type", "application/json")
                .build()

            val response = client.newCall(request).execute()
            val json = response.body?.string()

            if (response.isSuccessful && json != null) {
                val paymentResponse = gson.fromJson(json, PaymentResponse::class.java)
                if (paymentResponse.success) {
                    return@withContext paymentResponse.paymentStatus
                }
            }
            null
        } catch (e: IOException) {
            e.printStackTrace()
            null
        }
    }
}

// Usage example in Activity/Fragment
class CustomerDisplayActivity : AppCompatActivity() {
    private val apiClient = CartApiClient()
    private val scope = CoroutineScope(Dispatchers.Main + SupervisorJob())

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_customer_display)

        // Start monitoring cart
        startCartMonitoring()
    }

    private fun startCartMonitoring() {
        scope.launch {
            while (isActive) {
                try {
                    val cart = apiClient.getCart()
                    updateCartDisplay(cart)

                    val paymentStatus = apiClient.getPaymentStatus()
                    updatePaymentDisplay(paymentStatus)

                    // Poll every 2 seconds
                    delay(2000)
                } catch (e: Exception) {
                    e.printStackTrace()
                    // Handle error - maybe show offline message
                }
            }
        }
    }

    private fun updateCartDisplay(cart: Cart?) {
        if (cart == null) return

        if (cart.items.isEmpty()) {
            // Show empty cart message
            findViewById<TextView>(R.id.cartStatusText).text = "Cart is empty"
            findViewById<RecyclerView>(R.id.cartItemsRecyclerView).visibility = View.GONE
        } else {
            // Show cart items
            findViewById<TextView>(R.id.cartStatusText).text = "Current Order"
            findViewById<RecyclerView>(R.id.cartItemsRecyclerView).visibility = View.VISIBLE

            // Update total
            findViewById<TextView>(R.id.totalTextView).text =
                String.format("$%.2f", cart.total)

            // Update customer name if available
            cart.customer?.name?.let { name ->
                findViewById<TextView>(R.id.customerNameTextView).text = "Customer: $name"
            }
        }
    }

    private fun updatePaymentDisplay(paymentStatus: PaymentStatus?) {
        if (paymentStatus == null) return

        when (paymentStatus.status) {
            "processing" -> {
                findViewById<TextView>(R.id.paymentStatusText).text = "Processing payment..."
                // Show loading animation
            }
            "completed" -> {
                findViewById<TextView>(R.id.paymentStatusText).text = "Payment completed!"
                // Show success animation, maybe clear cart display
            }
            "failed" -> {
                findViewById<TextView>(R.id.paymentStatusText).text = "Payment failed"
                // Show error message
            }
            else -> {
                findViewById<TextView>(R.id.paymentStatusText).text = "Ready for payment"
            }
        }
    }

    override fun onDestroy() {
        super.onDestroy()
        scope.cancel()
    }
}